File: suwmp_be/src/main/java/com/example/suwmp_be/entity/Enterprise.java
package com.example.suwmp_be.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * Minimal Enterprise entity stub.
 * Full implementation will be provided by another team member.
 * This stub is only for compilation purposes to support EnterpriseCollector relationships.
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "enterprises")
@Data
public class Enterprise {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column
    private String description;

    @Column(nullable = false)
    private Float rating;

    @Column(name = "photo_url")
    private String photoUrl;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

}

File: suwmp_fe/src/types/enterprise.ts
export interface EnterpriseProfileGetResponse {
    id: number;
    name: string;
    description: string;
    rating: number;
    photoUrl: string;
    createdAt: string;
}

export type Enterprise = {
    name: string,
    photoUrl: string,
    description: string,
}

export interface EnterpriseProfileUpdateRequest {
    name: string;
    description: string;
    photoUrl: string;
}